pkg/api/mock/mock_api_implementation_test.go

```diff
@@ -0,0 +1,208 @@
+			err = a.Get(context.TODO(), searchVLAN)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(searchVLAN).To(Equal(vlan))
+			Expect(searchVLAN).To(Not(Equal(testVLAN)))
+		})
+
+		It("can List resources [ObjectChannel]", func() {
+			var oc types.ObjectChannel
+			err := a.List(context.TODO(), &clouddnsv1.Record{}, api.ObjectChannel(&oc))
+			Expect(err).ToNot(HaveOccurred())
+
+			var res []types.Object
+			for retriever := range oc {
+				var obj clouddnsv1.Record
+				err := retriever(&obj)
+				Expect(err).ToNot(HaveOccurred())
+				res = append(res, &obj)
+			}
+
+			Expect(res).To(ConsistOf(dnsRecords))
+		})
+
+		It("can List resources [Paged]", func() {
+			var pi types.PageInfo
+			err := a.List(context.TODO(), &clouddnsv1.Record{}, api.Paged(1, 1, &pi))
+			Expect(err).ToNot(HaveOccurred())
+
+			res := make([]types.Object, 0, pi.TotalItems())
+			var page []types.Object
+			for pi.Next(&page) {
+				res = append(res, page...)
+			}
+
+			Expect(res).To(ConsistOf(dnsRecords))
+		})
+
+	})
+
+	Context("Custom gomega matcher", Ordered, func() {
+		var a MockAPI
+		var identifiers []string
+		BeforeEach(func() {
+			a = NewMockAPI()
+
+			objects := []types.Object{
+				&clouddnsv1.Zone{Name: "mock-0.go-anxcloud.test"},
+				&clouddnsv1.Zone{Name: "mock-1.go-anxcloud.test"},
+				&clouddnsv1.Zone{Name: "mock-2.go-anxcloud.test"},
+
+				&clouddnsv1.Record{ZoneName: "mock-1.go-anxcloud.test", Name: "record-0"},
+				&clouddnsv1.Record{ZoneName: "mock-1.go-anxcloud.test", Name: "record-1"},
+
+				&clouddnsv1.Record{ZoneName: "mock-2.go-anxcloud.test", Name: "record-0"},
+				&clouddnsv1.Record{ZoneName: "mock-2.go-anxcloud.test", Name: "record-1"},
+				&clouddnsv1.Record{ZoneName: "mock-2.go-anxcloud.test", Name: "record-2"},
+			}
+
+			createdIdentifiers := make([]string, len(objects))
+
+			for i, obj := range objects {
+				err := a.Create(context.TODO(), obj)
+				Expect(err).To(Not(HaveOccurred()))
+				createdIdentifiers[i], err = api.GetObjectIdentifier(obj, true)
+				Expect(err).ToNot(HaveOccurred())
+			}
+
+			identifiers = createdIdentifiers
+		})
+
+		It("can match by type and fields", func() {
+			Expect(a).To(
+				ContainElementNTimes(
+					ObjectMatcher(&clouddnsv1.Record{ZoneName: "mock-1.go-anxcloud.test"}, "ZoneName"),
+					2,
+				),
+			)
+
+			Expect(a).To(
+				ContainElementNTimes(
+					ObjectMatcher(&clouddnsv1.Record{ZoneName: "mock-2.go-anxcloud.test"}, "ZoneName"),
+					3,
+				),
+			)
+		})
+
+		It("can match destroyed resources", func() {
+			for _, identifier := range identifiers[5:7] {
+				err := a.Destroy(context.TODO(), &clouddnsv1.Record{Identifier: identifier})
+				Expect(err).ToNot(HaveOccurred())
+			}
+
+			Expect(a).To(
+				ContainElementNTimes(
+					SatisfyAll(
+						ObjectMatcher(&clouddnsv1.Record{ZoneName: "mock-2.go-anxcloud.test"}, "ZoneName"),
+						ObjectDeleted(),
+					),
+					2,
+				),
+			)
+		})
+	})
+
+	//Context("Object tagging", Ordered, func() {
```

Comment:
Tests depend on #122

Reply:
Oh I see, we have a circular PR dependency